src/chemenv/tools/converters.py

```diff
@@ -0,0 +1,445 @@
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.get(url, timeout=self.timeout) as response:
+                    if response.status == 200:
+                        return await response.text()
+                    raise ValueError(
+                        f"PubChem API failed with status {response.status}"
+                    )
+            except Exception as e:
+                raise e
+
+    @backoff.on_exception(
+        backoff.expo,
+        (aiohttp.ClientError, asyncio.TimeoutError),
+        max_time=10,
+        logger=logger,
+    )
+    async def cactus(self) -> Optional[str]:
+        """
+        Query CACTUS API to get IUPAC name from SMILES.
+
+        Returns:
+            Optional[str]: IUPAC name if found, None if the query failed.
+
+        Raises:
+            aiohttp.ClientError: If the API request fails.
+            asyncio.TimeoutError: If the request times out.
+        """
+        inchi = Chem.MolToInchi(Chem.MolFromSmiles(self.smiles))
+        url = f"https://cactus.nci.nih.gov/chemical/structure/{inchi}/iupac_name"
+
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.get(url, timeout=self.timeout) as response:
+                    if response.status == 200:
+                        return await response.text()
+                    raise ValueError(f"CACTUS API failed with status {response.status}")
+            except Exception as e:
+                raise e
+
+    async def get_name(self) -> Optional[str]:
+        """
+        Query multiple chemical APIs in parallel to get IUPAC name.
+
+        Attempts to retrieve the IUPAC name by querying multiple chemical databases
+        concurrently (CACTUS and PubChem). Returns the first successful result.
+
+        Returns:
+            str: The IUPAC name of the chemical compound.
+
+        Raises:
+            ValueError: If no name could be found in any of the chemical databases.
+        """
+        tasks = [
+            self.cactus(),
+            self.pubchem(),
+        ]
+
+        for result in asyncio.as_completed(tasks):
+            try:
+                name = await result
+                if name:
+                    return name.strip()
+            except Exception:
+                continue
+
+        logger.error(f"Could not find name for {self.smiles}")
+        raise ValueError(f"Could not find name for {self.smiles}")
+
+
+def _smiles_to_selfies(smiles: str) -> str:
+    """
+    Takes a SMILES and return the SELFIES encoding.
+
+    Args:
+        smiles: SMILES string
+
+    Returns:
+        str: SELFIES of the input SMILES
+    """
+
+    return selfies.encoder(smiles.strip())
+
+
+def _smiles_to_deepsmiles(smiles: str) -> str:
+    """
+    Takes a SMILES and return the DeepSMILES encoding.
+
+    Args:
+        smiles: SMILES string
```

Comment:
a micro-nitpick is that you could still add the type annotations to some of those docstrings